.github/workflows/js-ci.yml

```
@@ -253,12 +255,12 @@ jobs:
         working-directory: js
 
       - name: Run Playwright tests
-        run: pnpm --fail-if-no-match --filter ${{ env.WORKSPACE }} test:integration
+        run: pnpm --fail-if-no-match --filter ${{ env.WORKSPACE }} test:integration -- --shard=${{ matrix.shardIndex }}/${{ matrix.shardTotal }}
         working-directory: js
 
       - name: Upload Playwright report
         uses: actions/upload-artifact@v4
-        if: always()
+        if: ${{ !cancelled() }}
```

I think you can omit the expression here.
```suggestion
        if: !cancelled()
```